qiskit/transpiler/_transpiler.py

```diff
@@ -88,92 +79,83 @@ def compile(circuits, backend,
+        if num_qubits == 1 or coupling_map == "all-to-all":
+            coupling_map = None
         job["config"]["coupling_map"] = coupling_map
-        # TODO: Jay: make config options optional for different backends
         job["config"]["basis_gates"] = basis_gates
-        if seed is None:
-            job["config"]["seed"] = None
-        else:
-            job["config"]["seed"] = seed
-
-        if skip_transpiler:  # Just return the qobj, without any transformation or analysis
-            job["config"]["layout"] = None
-            dag_circuit = DAGCircuit.fromQuantumCircuit(circuit)
-            job["compiled_circuit_qasm"] = dag_circuit.qasm(qeflag=True,
-                                                            eval_symbols=True)
-            job["compiled_circuit"] = DagUnroller(dag_circuit,
-                                                  JsonBackend(dag_circuit.basis)).execute()
-        else:
-            if initial_layout is None and not backend.configuration['simulator']:
-                # if coupling_map is not already satisfied, pick a good initial layout
-                # otherwise leave as q[i]->q[i]
-                if not _matches_coupling_map(circuit.data, backend.configuration['coupling_map']):
-                    initial_layout = _pick_best_layout(backend, num_qubits, circuit.get_qregs())
-
-            dag_circuit, final_layout = compile_circuit(
-                circuit,
-                basis_gates=basis_gates,
-                coupling_map=coupling_map,
-                initial_layout=initial_layout,
-                get_layout=True)
-            # Map the layout to a format that can be json encoded
-            list_layout = None
-            if final_layout:
-                list_layout = [[k, v] for k, v in final_layout.items()]
-            job["config"]["layout"] = list_layout
-
-            # the compiled circuit to be run saved as a dag
-            # we assume that compile_circuit has already expanded gates
-            # to the target basis, so we just need to generate json
-            json_circuit = DagUnroller(dag_circuit, JsonBackend(dag_circuit.basis)).execute()
-            job["compiled_circuit"] = json_circuit
-            # set eval_symbols=True to evaluate each symbolic expression
-            # TODO after transition to qobj, we can drop this
-            job["compiled_circuit_qasm"] = dag_circuit.qasm(qeflag=True,
-                                                            eval_symbols=True)
+        job["config"]["seed"] = seed
+
+        # step 3: populate the circuit `instructions` after compilation
+        # step 3a: circuit -> dag
+        dag_circuit = DAGCircuit.fromQuantumCircuit(circuit)
+
+        # TODO: move this inside the mapper pass
+        # pick a good initial layout if coupling_map is not already satisfied
+        # otherwise keep it as q[i]->q[i]
+        if (initial_layout is None and
+            coupling_map is not None and
```

Comment:
Please, double check, this condition does not appear in the [original `if`](https://github.com/QISKit/qiskit-core/pull/578/files#diff-d84cfcad92182447556156d4d8a14c69L134).

Reply:
already fixed in https://github.com/QISKit/qiskit-core/pull/578/commits/d50505dbb3cde63555637d5337ad019fb43ececc